lib/lists.ex

```diff
@@ -0,0 +1,582 @@
+defmodule Lists do
+  @moduledoc """
+  See https://sites.google.com/site/prologsite/prolog-problems/1
+  """
+
+  @doc """
+  Find the last element in a list.
+
+    iex> Lists.last([:a, :b, :c, :d])
+    :d
+  """
+  def last([element]), do: element
+  def last([_ | t]), do: last(t)
```

Comment:
Nice, as we discussed, might be worth handling the `[]` case, but throwing an error is probs fine

Reply:
Good call. @samphilipd was saying the convention was that mainly functions that end in `!` should throw exceptions, and others don't.

So I think I'll change this to return `{:ok, last_element}` if ok, `{:err, :empty_list}` so as to differentiate!